tools/release/get_version.py

```diff
@@ -0,0 +1,213 @@
+    return severity[min(trace_one_severity, trace_other_severity)]
+
+
+def get_needed_bump(history: list[str]) -> TraceabilityEnum:
+    """
+    Get what version number should be bumped using traceability postfixes
+
+    Breaking -> Major bump
+    New -> Minor bump
+    BugFix -> Patch bump
+    Infra -> No bump
+
+    Note: This yields a warning for commit messages are not in the format
+        Message (Traceability) (#PR)
+    """
+
+    needed_bump = TraceabilityEnum.INFRA
+    failed_category = []
+
+    # categorize all commits in the history
+    for commit_message in history:
+        *_, trace_str, pr = commit_message.rsplit(" ", 2)
+        try:
+            current_trace = TraceabilityEnum.parse(trace_str)
+        except ValueError:
+            # clean up the pr so that we can use it in the warning
+            pr = pr.replace("(", "").replace(")", "").replace("#", "")
+            failed_category.append(
+                FailedCategory(commit=commit_message, pr=pr)
+            )
+            continue
+        needed_bump = get_most_severe(needed_bump, current_trace)
+
+    # report commits that were not automatically categorized
+    if failed_category:
+        logger.warning("Failed to categorize:")
+    for failure in failed_category:
+        warning_failure_text = textwrap.dedent(
+            f"""
+            {failure.commit}
+            Check: https://github.com/canonical/checkbox/pull/{failure.pr}
+            """
+        ).strip()
+        logger.warning(warning_failure_text)
+
+    return needed_bump
+
+
+def add_dev_suffix(version: str, history_len: int):
+    return f"{version}-dev{history_len}"
+
+
+def get_bumped_version(version: str, needed_bump: TraceabilityEnum) -> str:
+    version_no_v = version.replace("v", "")
+    major, minor, patch = (int(n) for n in version_no_v.split("."))
+    if needed_bump == TraceabilityEnum.BREAKING:
+        major += 1
+        minor = 0
+        patch = 0
+    elif needed_bump == TraceabilityEnum.NEW:
+        minor += 1
+        patch = 0
+    elif needed_bump == TraceabilityEnum.BUGFIX:
+        patch += 1
+    else:
+        ...
+    return f"v{major}.{minor}.{patch}"
+
+
+def describe_bump(needed_bump: TraceabilityEnum) -> str:
+    if needed_bump == TraceabilityEnum.BREAKING:
+        return "major"
+    elif needed_bump == TraceabilityEnum.NEW:
+        return "minor"
+    elif needed_bump == TraceabilityEnum.BUGFIX:
+        return "patch"
+    return "none"
+
+
+def setup_logger(log_level: str):
+    """
+    Sets up the global logger to the provider log_level
+    """
+    global logger
+    logger = logging.getLogger(__name__)
+    logger.setLevel(getattr(logging, log_level))
+
+
+def get_cli_args(argv):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--log",
+        choices=["ERROR", "WARNING", "INFO"],
+        help="set the log level (default: %(default)s)",
+        default="WARNING",
```

Comment:
This is here because I've been using this script offline for a while and it is handy to just have the output for "dumb" redirections when using it by hand. Do you want me to remove the configuration or change the default to info?

Reply:
Whatever you prefer. One neat trick is to have -d or -v flags that store logging.DEBUG or logging.INFO respectively in an arg that gets logging.WARNING by default :)

But in this case it could be that `-d` makes the arg logging.DEBUG and the default being to warn.

Reply:
done